src/calendar-backend/Calendar.Api/Controllers/LectureController.cs

```diff
@@ -0,0 +1,91 @@
+﻿using AutoMapper;
+using Calendar.Api.DTOs;
+using Calendar.Api.DTOs.Create;
+using Calendar.Api.DTOs.Update;
+using Calendar.Api.Services.Interfaces;
+using Calendar.Mongo.Db.Models;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Calendar.Api.Controllers;
+
+[ApiController]
+[Route("v1/api/[controller]")]
+public class LectureController : ControllerBase
+{
+    private readonly ILectureService service;
+    private readonly IMapper mapper;
+    private readonly ILogger<LectureController> logger;
+
+    public LectureController(ILectureService service, IMapper mapper, ILogger<LectureController> logger)
+    {
+        this.service = service;
+        this.mapper = mapper;
+        this.logger = logger;
+    }
+
+    [HttpPost]
+    // [Authorize(Roles = "editor")]
+    public async Task<ActionResult<LectureDTO>> AddLecture([FromBody] CreateLectureDTO lecture)
+    {
+        if (lecture == null)
+        {
+            return BadRequest();
+        }
+
+        var result = await service.AddLectureAsync(mapper.Map<Lecture>(lecture));
+
+        return Ok(mapper.Map<LectureDTO>(result));
+    }
+
+    [HttpGet]
+    // [Authorize(Roles = "viewer,editor")]
+    public async Task<ActionResult<IEnumerable<LectureDTO>>> GetLecture(string lectureId)
+    {
+        var lecture = await service.GetLectureByIdAsync(lectureId);
+
+        if (lecture == null)
+        {
+            return NotFound();
+        }
+
+        return Ok(mapper.Map<IEnumerable<LectureDTO>>(lecture));
+    }
+
+    [HttpGet("{lectureId}")]
+    // [Authorize(Roles = "viewer,editor")]
+    public async Task<ActionResult<LectureDTO>> GetLectureById(string lectureId)
+    {
+        var lecture = await service.GetLectureByIdAsync(lectureId);
+
+        if (lecture == null)
+        {
+            return NotFound();
+        }
+
+        return Ok(mapper.Map<LectureDTO>(lecture));
+    }
+
+    [HttpPut("{lectureId}")]
+    // [Authorize(Roles = "editor")]
+    public async Task<ActionResult<LectureDTO>> EditLecture(string lectureId, [FromBody] UpdateLectureDTO lecture)
```

Comment:
![image](https://github.com/mhummel-artiso/lecture-calendar/assets/97592367/a46286ad-f4f3-4ca2-b956-1f2bec96fd2a)

![image](https://github.com/mhummel-artiso/lecture-calendar/assets/97592367/13b6833d-88b7-4e14-9190-fa3973e2b6bc)

Darauf achten, das wenn man nur eine Property ändert, z.B. Comment, das nicht andere Properties mit null überschrieben werden. Id nicht in Body, Id über Parameter.

Reply:
da, sollte das gesamte objekt mitgeschickt werden da sonst änderungen die von wo anderst gemacht werden zur hälfte überschrieben werden und somit fehlerhafte daten entstehen können

Reply:
Okay, verstehe. Macht Sinn.